tcms/telemetry/api.py

```diff
@@ -169,3 +169,59 @@ def _append_status_counts_to_result(count, result):
     for status in TestExecutionStatus.chart_status_names:
         status_count = count.get(status, 0)
         result.get(status).append(status_count)
+
+
+@rpc_method(name='Testing.test_case_health')
+def test_case_health(query=None):
+
+    if query is None:
+        query = {}
+
+    all_test_executions = TestExecution.objects.filter(**query)
+
+    test_executions = _get_count_for(all_test_executions)
+    passed_test_executions = _get_count_for(all_test_executions.filter(
+        status__name__in=TestExecutionStatus.passed_status_names))
+    failed_test_executions = _get_count_for(all_test_executions.filter(
+        status__name__in=TestExecutionStatus.failure_status_names))
+
+    data = {}
+    for te in test_executions:
+        data[te['case_id']] = {
+            'case_id': te['case_id'],
+            'case_summary': te['case__summary'],
+            'count': {
+                'all': te['count'],
+                'success': 0,
+                'fail': 0
+            }
+        }
+
+    for te in passed_test_executions:
+        data[te['case_id']]['count']['success'] = te['count']
+    for te in failed_test_executions:
+        data[te['case_id']]['count']['fail'] = te['count']
+
+    for key in data.copy():
+        if data[key]['count']['success'] == data[key]['count']['all']:
+            data.pop(key)
+
+    data = list(data.values())
+
+    data.sort(key=_sort_by_passing_rate)
+
+    result = []
+    for i in range(0, 20):
+        result.append(data[i])
```

Comment:
I think remove 211-215 for now. You will need 3 ranges of data points here, let's say the first 10 of each:

- pass rate == 0 - these have always failed and need immediate attention, this goes on top of the report
- pass rate > 0: 
   - closer to 100% - mostly fine but have been a bit flaky - means likely a test environment issue
   - closer to 0% - mostly failing with sometimes pass - means not very stable TC

^^^ because you need 3 data sets you may not be able to get everything with 1 ordering operation. Or maybe try order by their success rate and then try filtering by count based on the band values above. This should be doable in 1 loop with a few counters/if conditions.